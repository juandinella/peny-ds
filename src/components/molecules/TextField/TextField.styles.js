import { css } from 'styled-components'

import layout from '../../../styles/theme/layout'

export default css`
  width: 100%;
  position: relative;

  ${(props) =>
    props.isInline &&
    css`
      max-width: max-content;
    `}

  label {
    color: var(--color-neutral-600);
    display: block;
    margin-bottom: var(--spacing-xxs);
  }

  textarea {
    height: 120px;
    max-height: 120px;
    resize: none;
  }

  input,
  textarea {
    width: 100%;
    border: 0 none;
    border: 1px solid var(--color-neutral-600);
    font-size: var(--font-sm);
    transition: border-color var(--transition-timing-easing);
    padding: var(--spacing-xxs);
    box-sizing: border-box;
    border-radius: var(--radius-sm);

    &:disabled {
      background-color: inherit;
      border-color: var(--color-neutral-200);
      cursor: not-allowed;
      color: var(--color-neutral-200);
    }

    &:disabled + label {
      color: var(--color-neutral-200);
    }

    &:focus {
      outline: none;
      border-color: var(--color-primary-main);

      + label {
        font-size: var(--font-xxs);
        color: var(--color-primary-main);
      }
    }
  }

  .input-icon {
    position: absolute;
    right: var(--spacing-xs);
    top: 33px;

    svg {
      display: block;
    }
  }

  .input-feedback {
    margin-right: auto;
    font-size: var(--font-xs);
    color: var(--color-alert-error);
    margin-top: var(--spacing-xxxs);
  }

  ${(props) =>
    props.className === 'success' &&
    css`
      input,
      textarea {
        border: 1px solid var(--color-alert-success);
      }
    `}

  ${(props) =>
    props.className === 'error' &&
    css`
      input,
      textarea {
        border: 1px solid var(--color-alert-error);
      }
      input:focus,
      textarea:focus {
        border: 1px solid var(--color-alert-error);
      }
    `}


  ${(props) =>
    props.size &&
    css`
      input {
        font-size: var(--font-${(props) => props.size});
      }
    `}


  ${(props) =>
    props.size === 'md' &&
    css`
      input {
        &:focus {
          + label {
            font-size: var(--font-sm);
          }
        }

        &:not(:placeholder-shown) {
          + label {
            font-size: var(--font-sm);
          }
        }
      }
    `}

  ${(props) =>
    (props.size === 'xl' || props.size === 'xxl') &&
    css`
      input {
        font-size: var(--font-lg);

        @media screen and (min-width: ${() => layout.breakpoints.sm}) {
          font-size: var(--font-${(props) => props.size});
        }
      }
    `}

  ${(props) =>
    props.color &&
    css`
      input {
        color: var(--color-${(props) => props.color});
      }
    `}
`
