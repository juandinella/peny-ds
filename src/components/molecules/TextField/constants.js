export const options = {
  /* The type attr of input element defines information field */
  types: ['text', 'email', 'number', 'password'],
  sizes: ['sm', 'md', 'lg', 'xl', 'xxl'],
  colors: [
    'neutral-900',
    'neutral-800',
    'neutral-700',
    'neutral-600',
    'neutral-500',
    'neutral-400',
    'neutral-300',
    'neutral-200',
    'neutral-100',
    'neutral-50',
    'neutral-00',
    'primary-main',
    'secondary-main',
    'alert-error',
    'alert-success',
  ],
}
