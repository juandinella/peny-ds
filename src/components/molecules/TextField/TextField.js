import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import Skeleton from '../../atoms/Skeleton'
import styles from './TextField.styles'
import { options } from './constants'
import Utils from '../../../utils'

/**
 * Input fields let users enter and edit text.
 */

const ErrorSpan = ({ errors }) =>
  Array.isArray(errors) ? (
    errors.forEach((error) => <span>{error}</span>)
  ) : (
    <span className="input-feedback">{errors}</span>
  )

const Input = ({ textarea, ...props }) =>
  textarea ? <textarea {...props} /> : <input {...props} />

const TextField = ({
  type,
  value,
  name,
  isInline,
  onChange,
  onKeyDown,
  label,
  id,
  error,
  persistentError,
  icon,
  disabled,
  maxLength,
  pattern,
  children,
  shouldRender,
  placeholder,
  color,
  size,
  inputIcon,
  textarea,
}) => {
  const [touched, setTouched] = useState(false)
  const [focus, setFocus] = useState(false)

  const checkErrorFocus = () => (persistentError ? true : !focus)
  const checkErrorTouched = () => (persistentError ? true : touched)

  const inputChange = (e) => {
    if (pattern) {
      const regex = new RegExp(pattern)
      e.target.value = e.target.value.replace(regex, '')
    }
    if (onChange) onChange(e)
  }

  // Sets touched on first render if value is not empty
  useEffect(() => {
    if (!Utils.emptyValue(value)) setTouched(true)
  }, [])

  let className = ''

  if (checkErrorFocus()) {
    if (error && error.length > 0 && checkErrorTouched()) {
      className = 'error'
    } else {
      if (!Utils.emptyValue(value)) {
        className = 'success'
      }
    }
  }

  return (
    <Skeleton
      id={`${id}_skeleton`}
      show={!Utils.emptyValue(shouldRender) && !shouldRender()}
    >
      <div
        css={styles}
        className={className}
        isInline={isInline}
        size={size}
        color={color}
        inputIcon={inputIcon}
      >
        {label ? <label htmlFor={id}>{label}</label> : null}
        <Input
          textarea={textarea}
          type={type}
          value={value}
          name={name}
          maxLength={maxLength}
          id={id}
          data-testid={id}
          disabled={disabled}
          onChange={inputChange}
          placeholder={placeholder}
          onFocus={() => {
            setTouched(true)
            setFocus(true)
          }}
          onBlur={() => setFocus(false)}
          onKeyDown={onKeyDown}
        />
        {checkErrorFocus() &&
        checkErrorTouched() &&
        error &&
        error.length > 0 ? (
          <ErrorSpan errors={error} />
        ) : null}
        {checkErrorFocus() && !Utils.emptyValue(value) && icon ? (
          <span className="input-icon">
            {error && error.length > 0 ? (
              <svg
                xmlns="http://www.w3.org/2000/svg"
                width="24"
                height="24"
                viewBox="0 0 24 24"
                fill="none"
                stroke="#dc2626"
                strokeWidth="1.5"
                strokeLinecap="round"
                strokeLinejoin="round"
                className="feather feather-x-circle"
              >
                <circle cx="12" cy="12" r="10"></circle>
                <line x1="15" y1="9" x2="9" y2="15"></line>
                <line x1="9" y1="9" x2="15" y2="15"></line>
              </svg>
            ) : (
              <svg
                xmlns="http://www.w3.org/2000/svg"
                width="24"
                height="24"
                viewBox="0 0 24 24"
                fill="none"
                stroke="#22c55e"
                strokeWidth="1.5"
                strokeLinecap="round"
                strokeLinejoin="round"
                className="feather feather-check-circle"
              >
                <path d="M22 11.08V12a10 10 0 1 1-5.93-9.14"></path>
                <polyline points="22 4 12 14.01 9 11.01"></polyline>
              </svg>
            )}
          </span>
        ) : null}

        {children}
      </div>
    </Skeleton>
  )
}

TextField.displayName = 'TextField'

TextField.propTypes = {
  /** Function triggered on input change */
  onChange: PropTypes.func.isRequired,
  /** Function on key down */
  onKeyDown: PropTypes.func,
  /** Types of the input */
  type: PropTypes.oneOf(options.types),
  /** Input value */
  value: PropTypes.string,
  /** Input name */
  name: PropTypes.string,
  /** Specifies a Unique id to the element */
  id: PropTypes.string,
  /** Input label */
  label: PropTypes.string,
  /** This Boolean attribute to display an inline input */
  isInline: PropTypes.bool,
  /** Input Size */
  size: PropTypes.oneOf(options.sizes),
  /** Input Text Color */
  color: PropTypes.oneOf(options.colors),
  /** Input's placeholder.*/
  placeholder: PropTypes.string,
  /** This Boolean attribute prevents the user from interacting with the input */
  disabled: PropTypes.bool,
  /** String or array of strings to show if input value is wrong */
  error: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
  ]),
  /** Boolean attribute to display a success or error icon */
  icon: PropTypes.bool,
  /** Icon to display over the input */
  inputIcon: PropTypes.node,
  /**
   * Function that decides if the component should be rendered.
   *
   * @return {boolean} false if component should not be rendered and show skeleton instead. Component renders otherwise.
   */
  shouldRender: PropTypes.func,
  /** Input maxlenght*/
  maxLength: PropTypes.number,
  /* Whether input error should be shown always */
  persistentError: PropTypes.bool,
  /* Regex for input */
  pattern: PropTypes.string,
  /* Textarea */
  textarea: PropTypes.bool,
}

TextField.defaultProps = {
  type: 'text',
  value: '',
  name: '',
  label: '',
  id: '',
  placeholder: 'Â ',
  size: 'sm',
  color: 'neutral-900',
  isInline: false,
  onChange: () => null,
}

export default TextField
