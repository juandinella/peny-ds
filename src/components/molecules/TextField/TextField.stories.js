import TextField from './TextField'
import { options } from './constants'

import {
  getListTemplate,
  getOptionsArgTypes,
  getTemplateWithValue,
} from '../../../helpers/storybook'

const Template = getTemplateWithValue(TextField)
const ListTemplate = getListTemplate(TextField)

export default {
  title: 'Molecules/TextField',
  component: TextField,
  args: {
    label: 'Im an TextField',
    id: 'test-id',
    placeholder: 'This is a placeholder',
    size: 'sm',
    textarea: false,
  },
  argTypes: {
    type: getOptionsArgTypes(options.types),
    size: getOptionsArgTypes(options.sizes),
    color: getOptionsArgTypes(options.colors),
  },
}

export const Default = Template.bind({})

export const Value = Template.bind({})
Value.args = {
  value: 'Others mainly cakehole Thorin moldy facial.',
  id: 'value-id',
}

export const PatternOnlyNumbers = Template.bind({})
PatternOnlyNumbers.args = { pattern: '(?![0-9]).+', id: 'pattern-id' }

export const ValueSuccess = Template.bind({})
ValueSuccess.args = {
  value: 'Others mainly cakehole Thorin moldy facial.',
  icon: true,
  id: 'icon-id',
}

export const ValueError = Template.bind({})
ValueError.args = {
  value: 'Others mainly cakehole Thorin moldy facial.',
  error: 'Im an error',
  icon: true,
  id: 'error-id',
}

export const Types = ListTemplate.bind({})
Types.args = { items: options.types.map((type) => ({ type })) }

export const Inline = Template.bind({})
Inline.args = { isInline: true, id: 'inline-id' }

export const Sizes = ListTemplate.bind({})
Sizes.args = {
  value: 'text',
  hideLabel: true,
  id: 'size-id',
  items: options.sizes.map((size) => ({ size })),
}

export const Colors = ListTemplate.bind({})
Colors.args = {
  value: 'text',
  items: options.colors.map((color) => ({ color })),
}

export const Textarea = Template.bind({})
Textarea.args = {
  value:
    "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book",
  textarea: true,
  id: 'textarea-id',
}

export const Skeleton = Template.bind({})
Skeleton.args = { shouldRender: () => false }
