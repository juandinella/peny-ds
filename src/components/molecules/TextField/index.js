import TextField from './TextField'

export default TextField
export { options } from './constants'
