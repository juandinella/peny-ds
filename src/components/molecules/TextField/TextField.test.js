import React from 'react'

import { cleanup, render, fireEvent } from '@testing-library/react'

import TextField from './TextField'

import initStoryshots from '@storybook/addon-storyshots'

initStoryshots({
  storyKindRegex: /^Molecules\/TextField$/,
})

const componentProps = {
  id: 'this-id',
  label: 'Label',
  name: 'name',
  type: 'text',
  value: '1.00000000',
  onChange: jest.fn(),
  error: 'Error',
  icon: true,
  textarea: false,
}

afterEach(cleanup)

beforeEach(() => {
  componentProps.onChange.mockClear()
})

describe('<TextField />', () => {
  test('should render input correctly', () => {
    const { getByTestId, container } = render(<TextField {...componentProps} />)

    const textfield = getByTestId(componentProps.id)

    const input = container.querySelector('input')

    expect(textfield).toBeTruthy()
    expect(input).toBeTruthy()
  })

  test('should render textarea correctly', () => {
    const { getByTestId, container } = render(
      <TextField {...componentProps} textarea />
    )

    const textfield = getByTestId(componentProps.id)

    const textarea = container.querySelector('textarea')

    expect(textfield).toBeTruthy()
    expect(textarea).toBeTruthy()
  })

  test('should render skeleton', () => {
    const { queryByTestId } = render(
      <TextField {...componentProps} shouldRender={() => false} />
    )

    expect(queryByTestId(`${componentProps.id}_skeleton`)).toBeTruthy()
  })

  test('should render label correctly', () => {
    const { container } = render(<TextField {...componentProps} />)

    const label = container.querySelector('label')

    expect(label.innerHTML).toEqual(componentProps.label)
  })

  test('should render value correctly', () => {
    const { getByTestId } = render(<TextField {...componentProps} />)

    const textfield = getByTestId(componentProps.id)

    expect(textfield.value).toEqual(componentProps.value)
  })

  test('should render name correctly', () => {
    const { container } = render(<TextField {...componentProps} />)

    const input = container.querySelector('input')

    expect(input.getAttribute('name')).toEqual(componentProps.name)
  })

  test('should render type correctly', () => {
    const { getByTestId } = render(<TextField {...componentProps} />)

    const textfield = getByTestId(componentProps.id)

    expect(textfield.type).toEqual(componentProps.type)
  })

  test('should render error correctly', () => {
    const { container, getByTestId } = render(<TextField {...componentProps} />)

    const input = getByTestId(componentProps.id)

    expect(container.querySelector('.input-feedback')).toBeTruthy()

    expect(container.querySelector('.input-feedback')).toBeTruthy()

    input.focus()

    expect(container.querySelector('.input-icon')).toBeFalsy()

    expect(container.querySelector('.input-icon')).toBeFalsy()
  })

  test('should change input correctly', () => {
    const { getByTestId } = render(<TextField {...componentProps} />)

    fireEvent.change(getByTestId(componentProps.id), {
      target: { value: '11' },
    })

    expect(componentProps.onChange).toHaveBeenCalledTimes(1)
  })
})
