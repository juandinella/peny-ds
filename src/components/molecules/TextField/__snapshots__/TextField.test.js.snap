// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/TextField Colors 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-x8vv3i-0 ktfbvR success"
    color="neutral-900"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
  <div
    className="sc-x8vv3i-0 fqziXk success"
    color="neutral-800"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
  <div
    className="sc-x8vv3i-0 chovWH success"
    color="neutral-700"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
  <div
    className="sc-x8vv3i-0 JMSqW success"
    color="neutral-600"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
  <div
    className="sc-x8vv3i-0 hzVIYR success"
    color="neutral-500"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
  <div
    className="sc-x8vv3i-0 cxpQAk success"
    color="neutral-400"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
  <div
    className="sc-x8vv3i-0 iCPAKf success"
    color="neutral-300"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
  <div
    className="sc-x8vv3i-0 hfnXeu success"
    color="neutral-200"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
  <div
    className="sc-x8vv3i-0 cGbent success"
    color="neutral-100"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
  <div
    className="sc-x8vv3i-0 jBaqol success"
    color="neutral-50"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
  <div
    className="sc-x8vv3i-0 iJyrjc success"
    color="neutral-00"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
  <div
    className="sc-x8vv3i-0 ciBEBy success"
    color="primary-main"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
  <div
    className="sc-x8vv3i-0 gQVzSc success"
    color="secondary-main"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
  <div
    className="sc-x8vv3i-0 inNEnb success"
    color="alert-error"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
  <div
    className="sc-x8vv3i-0 jWkjAg success"
    color="alert-success"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
</div>
`;

exports[`Storyshots Molecules/TextField Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-x8vv3i-0 doXjBO"
    color="neutral-900"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`Storyshots Molecules/TextField Inline 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-x8vv3i-0 dcGLDM"
    color="neutral-900"
    size="sm"
  >
    <label
      htmlFor="inline-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="inline-id"
      id="inline-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`Storyshots Molecules/TextField Pattern Only Numbers 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-x8vv3i-0 doXjBO"
    color="neutral-900"
    size="sm"
  >
    <label
      htmlFor="pattern-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="pattern-id"
      id="pattern-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`Storyshots Molecules/TextField Sizes 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-x8vv3i-0 ktfbvR success"
    color="neutral-900"
    size="sm"
  >
    <label
      htmlFor="size-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="size-id"
      id="size-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
  <div
    className="sc-x8vv3i-0 fULjsH success"
    color="neutral-900"
    size="md"
  >
    <label
      htmlFor="size-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="size-id"
      id="size-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
  <div
    className="sc-x8vv3i-0 hrfMNE success"
    color="neutral-900"
    size="lg"
  >
    <label
      htmlFor="size-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="size-id"
      id="size-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
  <div
    className="sc-x8vv3i-0 bdJKmy success"
    color="neutral-900"
    size="xl"
  >
    <label
      htmlFor="size-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="size-id"
      id="size-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
  <div
    className="sc-x8vv3i-0 dTVkai success"
    color="neutral-900"
    size="xxl"
  >
    <label
      htmlFor="size-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="size-id"
      id="size-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="text"
    />
  </div>
</div>
`;

exports[`Storyshots Molecules/TextField Skeleton 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-kjnynt-0 jfZhXD"
    data-testid="test-id_skeleton"
    id="test-id_skeleton"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <div
      className="sc-x8vv3i-0 doXjBO"
      color="neutral-900"
      size="sm"
    >
      <label
        htmlFor="test-id"
      >
        Im an TextField
      </label>
      <input
        data-testid="test-id"
        id="test-id"
        name=""
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="This is a placeholder"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/TextField Textarea 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-x8vv3i-0 ktfbvR success"
    color="neutral-900"
    size="sm"
  >
    <label
      htmlFor="textarea-id"
    >
      Im an TextField
    </label>
    <textarea
      data-testid="textarea-id"
      id="textarea-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book"
    />
  </div>
</div>
`;

exports[`Storyshots Molecules/TextField Types 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-x8vv3i-0 doXjBO"
    color="neutral-900"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value=""
    />
  </div>
  <div
    className="sc-x8vv3i-0 doXjBO"
    color="neutral-900"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="email"
      value=""
    />
  </div>
  <div
    className="sc-x8vv3i-0 doXjBO"
    color="neutral-900"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="number"
      value=""
    />
  </div>
  <div
    className="sc-x8vv3i-0 doXjBO"
    color="neutral-900"
    size="sm"
  >
    <label
      htmlFor="test-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="test-id"
      id="test-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="password"
      value=""
    />
  </div>
</div>
`;

exports[`Storyshots Molecules/TextField Value 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-x8vv3i-0 ktfbvR success"
    color="neutral-900"
    size="sm"
  >
    <label
      htmlFor="value-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="value-id"
      id="value-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="Others mainly cakehole Thorin moldy facial."
    />
  </div>
</div>
`;

exports[`Storyshots Molecules/TextField Value Error 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-x8vv3i-0 ktfbvR success"
    color="neutral-900"
    size="sm"
  >
    <label
      htmlFor="error-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="error-id"
      id="error-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="Others mainly cakehole Thorin moldy facial."
    />
    <span
      className="input-icon"
    >
      <svg
        className="feather feather-x-circle"
        fill="none"
        height="24"
        stroke="#dc2626"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="1.5"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <line
          x1="15"
          x2="9"
          y1="9"
          y2="15"
        />
        <line
          x1="9"
          x2="15"
          y1="9"
          y2="15"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`Storyshots Molecules/TextField Value Success 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-x8vv3i-0 ktfbvR success"
    color="neutral-900"
    size="sm"
  >
    <label
      htmlFor="icon-id"
    >
      Im an TextField
    </label>
    <input
      data-testid="icon-id"
      id="icon-id"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      type="text"
      value="Others mainly cakehole Thorin moldy facial."
    />
    <span
      className="input-icon"
    >
      <svg
        className="feather feather-check-circle"
        fill="none"
        height="24"
        stroke="#22c55e"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="1.5"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
        />
        <polyline
          points="22 4 12 14.01 9 11.01"
        />
      </svg>
    </span>
  </div>
</div>
`;
